client/src/components/Connections/components/Results.js

```diff
@@ -1,25 +1,43 @@
 // @flow
 import React from 'react'
 import {compose} from 'redux'
-import {branch, renderComponent} from 'recompose'
+import {branch, renderComponent, withState, withHandlers} from 'recompose'
+import {Collapse, Button} from 'reactstrap'
 import ConnectionWrapper, {type ConnectionProps} from '../dataWrappers/ConnectionWrapper'
 import EntityWrapper, {type EntityProps} from '../dataWrappers/EntityWrapper'
 import EntitySearchWrapper, {type EntitySearchProps} from '../dataWrappers/EntitySearchWrapper'
 import InfoLoader from './InfoLoader'
 import BeforeResults from './BeforeResults'
 import Subgraph from './Subgraph'
+import './Results.css'
 
-type Props = EntitySearchProps & EntityProps & ConnectionProps
+type StateProps = {
+  graphShown: boolean,
+  toggleGraphShown: () => void,
+}
+
+type Props = EntitySearchProps & EntityProps & ConnectionProps & StateProps
 
 const Results = (props: Props) => (
   <div>
-    {props.showGraph ? <Subgraph preloadNodes {...props} /> : null}
+    <div className="showGraphButtonWrap">
+      <Button color="primary" onClick={props.toggleGraphShown} className="showGraphButton">
+        {props.graphShown ? 'Skryť graf' : 'Zobraziť graf'}
+      </Button>
+    </div>
+    <Collapse isOpen={props.graphShown}>
+      <Subgraph preloadNodes {...props} />
+    </Collapse>
     {props.connections.map((connEid) => <InfoLoader key={connEid} eid={connEid} hasConnectLine />)}
   </div>
 )
 
 export default compose(
   EntitySearchWrapper,
+  withState('graphShown', 'setGraphShown', false),
+  withHandlers({
+    toggleGraphShown: ({setGraphShown, graphShown}) => () => setGraphShown(!graphShown),
```

Comment:
Add type `StateProps` to function args.

Reply:
I am not sure what you mean. I am defining `toggleGraphShown` prop here, `setGraphShown` is another method, that isn't included in `StateProps`.

Reply:
Oh, you are right ... still it would be nice to add types for those args.